airbyte-integrations/connectors/source-gitlab/source_gitlab/util.py

```diff
@@ -0,0 +1,21 @@
+import requests
+
+def get_group_list(**kwargs):
+    headers = kwargs["authenticator"].get_auth_header()
+
+    ids = []
+
+    r = requests.get(f'https://{kwargs["api_url"]}/api/v4/groups?page=1&per_page=50', headers=headers)
+    results = r.json()
+    items = map(lambda i: i['full_path'].replace('/', '%2f'), results)
+    ids.extend(items)
+
+    while 'X-Next-Page' in r.headers and r.headers['X-Next-Page'] != '':
+        next_page = r.headers['X-Next-Page']
+        per_page = r.headers['X-Per-Page']
+        r = requests.get(f'https://{kwargs["api_url"]}/api/v4/groups?page={next_page}&per_page={per_page}', headers=headers)
+        results = r.json()
+        items = map(lambda i: i['full_path'].replace('/', '%2f'), results)
+        ids.extend(items)
+
+    return ids
```

Comment:
```suggestion
    headers = kwargs["authenticator"].get_auth_header()

    ids = []
    has_next = True
    # First request params
    per_page = 50
    next_page = 1

    while has_next:
        r = requests.get(f'https://{kwargs["api_url"]}/api/v4/groups?page={next_page}&per_page={per_page}', headers=headers)
        next_page = r.headers.get('X-Next-Page')
        per_page = r.headers.get('X-Per-Page')
        results = r.json()
        items = map(lambda i: i['full_path'].replace('/', '%2f'), results)
        ids.extend(items)
        has_next = 'X-Next-Page' in r.headers and r.headers['X-Next-Page'] != ''

    return ids
```

Reply:
Refactored. This was admittedly messy :)